src/components/dashboard/expense-list-section/expense-list-section.tsx

```diff
@@ -1,29 +1,69 @@
-import { useState } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import { SortOrder } from "../../../@types/sortOrderTypes";
 import ExpenseListToolbar from "./expense-list-toolbar/expense-list-toolbar";
-import { ExpenseList } from "./expense-list";
-import { useExpenseLists } from "./useExpenseLists";
-import { SuspenseComponent } from "../../shared";
+import { ExpenseList } from "./expense-list/expense-list";
 import "./expense-list-section.css";
+import { ExpenseLIstType } from "../../../@types/expense-list-prop";
+import { useExpenseLists } from "./useExpenseLists";
+import { DataLoader } from "../../shared/DataLoader";
 
 export const ExpenseListSection = () => {
   const [sortOrder, setSortOrder] = useState<SortOrder>("asc");
-  const { data: section, isLoading, error } = useExpenseLists(0, 5, sortOrder);
+  const [offset, setOffset] = useState(0);
+  const [lists, setLists] = useState<ExpenseLIstType[]>([]);
+  const {
+    data: section,
+    isLoading,
+    error,
+  } = useExpenseLists(offset, 5, sortOrder);
+
+  const updateLists = useCallback(() => {
+    if (section?.data) {
+      setLists((prevLists) =>
+        offset === 0 ? section.data : [...prevLists, ...section.data]
```

Comment:
same solution as before, no need to branch.
if you do ```setList(prev => [...prev, section.data])``` you'll have the correct behavior.
when offset is 0, you set list to [].